uaclient/files.py

```diff
@@ -0,0 +1,296 @@
+import json
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Optional
+
+from uaclient import defaults, event_logger, schemas, util
+
+PRIVATE_SUBDIR = "private"
+
+event = event_logger.get_event_logger()
+LOG = logging.getLogger(__name__)
+
+
+class UAFile:
+    _root_directory = defaults.DEFAULT_DATA_DIR
+    _file_name = "file"
+
+    def __init__(
+        self,
+        root_directory: Optional[str] = None,
```

Comment:
Can't the default value here be `defaults.DEFAULT_DATA_DIR` ? Because then we would need need the `if` check later

Reply:
Makes sense.